modules/indexer/internal/bleve/util_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package bleve
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBleveIndexAndSearch(t *testing.T) {
+	scenarios := []struct {
+		Input     string
+		Fuzziness int
+	}{
+		{
+			Input:     "",
+			Fuzziness: 0,
+		},
+		{
+			Input:     "Avocado",
+			Fuzziness: 1,
+		},
+		{
+			Input:     "Geschwindigkeit",
+			Fuzziness: 2,
+		},
+		{
+			Input:     "non-exist",
+			Fuzziness: 0,
+		},
+		{
+			Input:     "갃갃갃",
+			Fuzziness: 0,
```

Comment:
I have no idea what these numbers mean.
Perhaps we should use constants instead?

Reply:
These are the expected fuzzyness for each string. However, I'm not quite sure on how to name the constant :( (if you guys have any ideias, I'm open)

A little more context, the method `GuessFuzzinessByKeyword` already [exists on gitea codebase](https://github.com/go-gitea/gitea/blob/main/modules/indexer/internal/bleve/util.go#L51) but lacks a unit tests. As of today, it calculates the fuzziness based on the size of the whole search criterion (unless some non-standard characters are used, in such a case, it defaults to zero). 

What I did was simply calculate the fuziness based on the terms instead of the whole string. This prevent that a criterion composed of a bunch of small words to have high degree of fuzziness. A fuzziness degree was rendering bleve too leninent and making it return results that would make much sense (e.g., [this](https://github.com/go-gitea/gitea/blob/05a5090c34530d0bcfd62776cbfd42f6da3386bf/modules/indexer/code/indexer_test.go#L80) scenario would return some results when it should return none)

Reply:
Can you perhaps add that as docs as well to the method?